frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -1335,79 +1431,45 @@ export const multitabEditorLogic = kea<multitabEditorLogicType>([
             const createQueryTab = async (): Promise<void> => {
                 if (searchParams.open_view) {
                     // Open view
-                    const viewId = searchParams.open_view
-
-                    if (values.dataWarehouseSavedQueries.length === 0) {
-                        await dataWarehouseViewsLogic.asyncActions.loadDataWarehouseSavedQueries()
-                    }
-
-                    const view = values.dataWarehouseSavedQueries.find((n) => n.id === viewId)
-                    if (!view) {
-                        lemonToast.error('View not found')
-                        return
+                    const view = dataWarehouseViewsLogic.values.dataWarehouseSavedQueryMapById[searchParams.open_view]
+                    if (view) {
+                        const queryToOpen = searchParams.open_query ? searchParams.open_query : view.query.query
+                        actions.editView(queryToOpen, view)
+                        tabAdded = true
                     }
-
-                    const queryToOpen = searchParams.open_query ? searchParams.open_query : view.query.query
-
-                    actions.editView(queryToOpen, view)
-                    tabAdded = true
-                    router.actions.replace(router.values.location.pathname)
```

Comment:
Any reason for the router deletion here? We want to lose the query parms once the editor has loaded